model_analyzer/record/record.py

```diff
@@ -24,14 +24,34 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 import time
+from abc import ABC, abstractmethod
 
 
-class Record:
+class Record(ABC):
     """
     This class is used for representing monitoring
     records
     """
 
-    def __init__(self, name):
-        self.time = time.time()
-        self.name = name
+    @abstractmethod
+    def header(self):
+        """
+        Returns
+        -------
+        str
+            The full name of the
+            metric.
+        """
+
+    @abstractmethod
+    def value(self):
```

Comment:
You should set this to return a float.

Reply:
Should we restrict the value of a record to always be a float? We may need to have them be strings or custom types. 
I suppose I could change it because all we have now are floats.